mbuild/recipes/silica_interface.py

```diff
@@ -0,0 +1,143 @@
+        self.add(interface)
+
+    def _strip_stray_atoms(self):
+        """ Remove stray atoms and surface pieces """
+
+        major_component = max(nx.connected_components(self.bond_graph), key=len)
+        for atom in list(self.particles()):
+            if not self.bond_graph.has_node(atom):
+                self.remove(atom)
+            elif atom not in major_component:
+                self.remove(atom)
+
+    def _bridge_dangling_Os(self, oh_density, thickness):
+        """ Create Si-O-Si bridges on the surface to yield the desired
+            density of reactive surface sites
+        """
+
+        area = self.periodicity[0] * self.periodicity[1]
+        target = int(oh_density * area)
+
+        dangling_Os = []
+        for atom in list(self.particles()):
+            if atom.name == 'O' and atom.pos[2] > thickness and len(self.bond_graph.neighbors(atom)) == 1:
+                dangling_Os.append(atom)
+
+        n_bridges = int((len(dangling_Os) - target) / 2)
+
+        for _ in range(n_bridges):
+            bridged = False
+            while not bridged:
+                O1 = choice(dangling_Os)
+                Si1 = self.bond_graph.neighbors(O1)[0]
+                for O2 in dangling_Os:
+                    if O2 == O1:
+                        continue
+                    Si2 = self.bond_graph.neighbors(O2)[0]
+                    if Si1 == Si2:
+                        continue
+                    if any(neigh in self.bond_graph.neighbors(Si2) for neigh in self.bond_graph.neighbors(Si1)):
+                        continue
+                    r = self.min_periodic_distance(Si1.pos, Si2.pos)
+                    if r < 0.45:
+                        bridged = True
+                        self.add_bond((O1, Si2))
+                        dangling_Os.remove(O1)
+                        dangling_Os.remove(O2)
+                        self.remove(O2)
+                        break
+
+    def _identify_surface_sites(self, thickness):
+        """ Label surface sites and add ports above them """
+
+        for atom in list(self.particles()):
+            if len(self.bond_graph.neighbors(atom)) == 1:
+                if atom.name == 'O' and atom.pos[2] > thickness:
+                    atom.name = 'OS'
+                    port = mb.Port(anchor=atom)
+                    mb.rotate_around_x(port, np.pi/2)
+                    mb.translate(port, atom.pos + np.array([0.0, 0.0, 0.1]))
+                    self.add(port, "port_{}".format(len(self.referenced_ports())))
+
+    def _adjust_stoichiometry(self, O_buffer):
+        """ Remove O's from the underside of the surface to yield a 2:1 Si:O ratio """
+
+        O_buffer = 0.275
+        num_O = len(list(self.particles_by_name('O')))
+        num_Si = len(list(self.particles_by_name('Si')))
+        n_deletions = num_O - 2*num_Si
+
+        bottom_Os = []
+        for atom in list(self.particles()):
```

Comment:
`list()` casting unnecessary here too and you could also write this as a list comprehension.